libraries/AP_DDS/AP_DDS_External_Odom.cpp

```diff
@@ -0,0 +1,72 @@
+
+
+#include "AP_DDS_External_Odom.h"
+#include "AP_DDS_Type_Conversions.h"
+
+#if AP_DDS_VISUALODOM_ENABLED
+
+#include <AP_VisualOdom/AP_VisualOdom.h>
+#include <GCS_MAVLink/GCS.h>
+
+void AP_DDS_External_Odom::handle_external_odom(const tf2_msgs_msg_TFMessage& msg)
+{
+    auto *visual_odom = AP::visualodom();
+    if (visual_odom == nullptr) {
+        return;
+    }
+
+    for (size_t i = 0; i < msg.transforms_size; i++) {
+        const auto ros_transform_stamped = msg.transforms[i];
+        if (is_odometry_frame(ros_transform_stamped)) {
+            const uint64_t remote_time_us {AP_DDS_Type_Conversions::time_u64_micros(ros_transform_stamped.header.stamp)};
+
+            Vector3f ap_position {};
```

Comment:
default constructor zeros, so {} not needed

Reply:
Done 5425deafcdddee4a3c3649561edb413f96ba0833